File: gpfksdlrn/level1/햄버거_만들기.js
// https://school.programmers.co.kr/learn/courses/30/lessons/133502

function solution(ingredient) {
  const stack = [];
  let count = 0;

  for (const item of ingredient) {
    stack.push(item);
    if (
      stack.length >= 4 &&
      stack[stack.length - 4] === 1 &&
      stack[stack.length - 3] === 2 &&
      stack[stack.length - 2] === 3 &&
      stack[stack.length - 1] === 1
    ) {
      stack.length -= 4;
      count++;
    }
  }

  return count;
}

File: gnsals0904/level1/햄버거_만들기.java
class Solution {
    public int solution(int[] ingredient) {
        int answer = 0;

        int[] stack = new int[ingredient.length];
        int top = 0;

        for (int item : ingredient) {
            stack[top++] = item;

            if (top >= 4
                    && stack[top - 4] == 1
                    && stack[top - 3] == 2
                    && stack[top - 2] == 3
                    && stack[top - 1] == 1) {

                top -= 4;
                answer++;
            }
        }

        return answer;
    }
}
